### PR TITLE
AZP: Test ucx binary compatibility between OS versions.

### PR DESCRIPTION
**Pull request by petro-rudenko**
## What
Build ucx in centos7 container and run ucx_info and perftest on ubuntu20

### CODE REVIEW

File: buildlib/azure-pipelines-pr.yml [hunk 1]
```diff
@@ -252,19 +252,63 @@ stages:
...
+        dependsOn: Distros
+        displayName: UCX tests in containers
+        pool:
+          name: MLNX
+          demands:
+          - ucx_docker -equals yes
+        strategy:
+          matrix:
+            ubuntu20:
+              CONTAINER: ubuntu2004
```
Comment (avildema) [inline buildlib/azure-pipelines-pr.yml:280]: You build on Centos and launch on Ubuntu? is it correct?

Comment (petro-rudenko) [reply buildlib/azure-pipelines-pr.yml:280]: Yes that's the goal of the test

File: buildlib/azure-pipelines-pr.yml [hunk 2]
```diff
@@ -252,19 +252,63 @@ stages:
...
+            displayName: 'Download Build Artifacts'
+            inputs:
+              artifactName: ucx_bin_$(Build.BuildId)
+              downloadPath: $(System.DefaultWorkingDirectory)
+          - bash: chmod u+rwx $(System.DefaultWorkingDirectory)/ucx_bin_$(Build.BuildId) -R
+          - bash: |
               set +eE
               set -x
-              ./src/tools/info/ucx_info -e -u t 2>&1 | tee info.txt
+              export LD_LIBRARY_PATH=$(System.DefaultWorkingDirectory)/ucx_bin_$(Build.BuildId)/install/lib:$LD_LIBRARY_PATH
```
Comment (yosefe) [inline buildlib/azure-pipelines-pr.yml:293]: `cd $(System.DefaultWorkingDirectory)/ucx_bin_$(Build.BuildId)/install`

File: buildlib/azure-pipelines-pr.yml [hunk 3]
```diff
@@ -252,19 +252,63 @@ stages:
...
               exit 0;
             displayName: Test ucx_info
+          - bash: |
+              set +eE
+              set -x
+              export LD_LIBRARY_PATH=$(System.DefaultWorkingDirectory)/ucx_bin_$(Build.BuildId)/install/lib:$LD_LIBRARY_PATH
+              $(System.DefaultWorkingDirectory)/ucx_bin_$(Build.BuildId)/install/bin/ucx_perftest -t tag_bw &
+              sleep 1;
+              $(System.DefaultWorkingDirectory)/ucx_bin_$(Build.BuildId)/install/bin/ucx_perftest -t tag_bw localhost 2>&1 | tee perf.txt
+              grep "Final:" perf.txt
```
Comment (yosefe) [inline buildlib/azure-pipelines-pr.yml:309]: add wait later to wait for 1st background task

File: buildlib/azure-pipelines-pr.yml [hunk 4]
```diff
@@ -252,19 +252,63 @@ stages:
...
               retVal=$?
               if [ $retVal -eq 0 ]; then
                   exit 1;
               fi
               exit 0;
             displayName: Test ucx_info
+          - bash: |
+              set +eE
+              set -x
+              export LD_LIBRARY_PATH=$(System.DefaultWorkingDirectory)/ucx_bin_$(Build.BuildId)/install/lib:$LD_LIBRARY_PATH
```
Comment (yosefe) [inline buildlib/azure-pipelines-pr.yml:305]: `cd $(System.DefaultWorkingDirectory)/ucx_bin_$(Build.BuildId)/install`

File: buildlib/azure-pipelines-pr.yml [hunk 5]
```diff
@@ -300,12 +301,12 @@ stages:
               exit 0;
             displayName: Test ucx_info
           - bash: |
-              set +eE
-              set -x
-              export LD_LIBRARY_PATH=$(System.DefaultWorkingDirectory)/ucx_bin_$(Build.BuildId)/install/lib:$LD_LIBRARY_PATH
-              $(System.DefaultWorkingDirectory)/ucx_bin_$(Build.BuildId)/install/bin/ucx_perftest -t tag_bw &
+              set +eE -x
```
Comment (yosefe) [inline buildlib/azure-pipelines-pr.yml:304]: why ignore errors here?

File: buildlib/azure-pipelines-pr.yml (hunk 2 above)
Comment (yosefe) [reply buildlib/azure-pipelines-pr.yml:291]: move it after "cd" and export LD_LIBRARY_PATH=$PWD/lib:$LD_LIBRARY_PATH

File: buildlib/azure-pipelines-pr.yml [hunk 6]
```diff
@@ -252,19 +252,64 @@ stages:
...
+            displayName: 'Download Build Artifacts'
+            inputs:
+              artifactName: ucx_bin_$(Build.BuildId)
+              downloadPath: $(System.DefaultWorkingDirectory)
+          - bash: chmod u+rwx $(System.DefaultWorkingDirectory)/ucx_bin_$(Build.BuildId) -R
+          - bash: |
+              export LD_LIBRARY_PATH=$(System.DefaultWorkingDirectory)/ucx_bin_$(Build.BuildId)/install/lib:$LD_LIBRARY_PATH
+              cd $(System.DefaultWorkingDirectory)/ucx_bin_$(Build.BuildId)/install
+          - bash: |
+              set +eE -x
```
Comment (yosefe) [inline buildlib/azure-pipelines-pr.yml:294]: why ignore errors here?

File: buildlib/azure-pipelines-pr.yml [hunk 7]
```diff
@@ -252,19 +252,64 @@ stages:
...
             displayName: Test ucx_info
+          - bash: |
+              set +eE -x
+              ./bin/ucx_perftest -t tag_bw &
+              server_pid=$!
+              sleep 1;
+              ./bin/ucx_perftest -t tag_bw localhost 2>&1 | tee perf.txt
+              wait $server_pid
+              grep "Final:" perf.txt
+              exit $?;
```
Comment (yosefe) [inline buildlib/azure-pipelines-pr.yml:311]: why needed?

File: buildlib/azure-pipelines-pr.yml [hunk 8]
```diff
@@ -288,27 +288,31 @@ stages:
...
               grep -i error info.txt
               retVal=$?
               if [ $retVal -eq 0 ]; then
                   exit 1;
               fi
               exit 0;
             displayName: Test ucx_info
+            workingDirectory: $(System.DefaultWorkingDirectory)/ucx_bin_$(Build.BuildId)/install
+            env: 
+              LD_LIBRARY_PATH: $(System.DefaultWorkingDirectory)/ucx_bin_$(Build.BuildId)/install/lib:$LD_LIBRARY_PATH
```
Comment (yosefe) [inline buildlib/azure-pipelines-pr.yml:302]: 👍

File: buildlib/azure-pipelines-pr.yml (hunk 7 above)
Comment (yosefe) [reply buildlib/azure-pipelines-pr.yml:311]: i mean, if we just do "set -Eex", i'd expect grep failure will lead to step failed. no need also have exit command

**Review by yosefe — APPROVED** [review]

**Comment by petro-rudenko** [comment]
squashed

**Pull request merged**